src/docTree.ts

```diff
@@ -34,11 +34,66 @@ export function getDocsRootNode(): Promise<DocNode> {
   if (getDocsRootNodeCache) {
     return getDocsRootNodeCache;
   }
-  getDocsRootNodeCache = getDocsRootNodeUncached();
+  getDocsRootNodeCache = getDocsRootNodeCached();
   return getDocsRootNodeCache;
 }
 
-async function getDocsRootNodeUncached(): Promise<DocNode> {
+function reconstructParentReferences(node: DocNode, parent?: DocNode): void {
+  if (parent) {
+    node.parent = parent;
+  }
+  node.children.forEach(child => reconstructParentReferences(child, node));
+}
+
+async function getDocsRootNodeCached(): Promise<DocNode> {
+  // In development, scan filesystem for hot reloading
+  // In production (including CI builds), load from pre-computed JSON
+  if (process.env.NODE_ENV === 'development') {
+    return getDocsRootNodeUncached();
+  }
```

Comment:
Not very efficient to check this at every invocation of the function as env variable value would never change across the lifetime of the app. A more efficient approch would be to define the function differently based on the env var value. Something along these lines:

```ts

const getDocsRootNodeCached = process.env.NODE_ENV === 'production' ? async () => {

...
...
} : getDocsRootNodeUncached;

```
Also note that I check for `=== 'production'` instead of `=== 'development'` as during development, `process.env.NODE_ENV` is usually empty or undefined.